### DISCUSSION TITLE
Remove cookie

### DISCUSSION THREAD

**Discussion by DZ8540** [adonisjs/core#3532] [Help] [answered]
Hello there! How i can do remove cookie for other web site?

I do this
```js
response.cookie('cookieKey', null)
```

or this
```js
response.cookie('cookieKey', undefined)
```

But it doesn't work(

**Comment by thetutlage** [comment, marked as answer]
Use `response.clearCookie('cookieKey')`

**Reply by DZ8540** [reply]
Best. As always, thanks!

**Reply by DZ8540** [reply]
But why this method is not here?
https://docs.adonisjs.com/guides/cookies

**Reply by thetutlage** [reply]
Yup, seems like missing in docs. Can you please create a PR to add it to the docs?

**Reply by DZ8540** [reply]
Yes, i do this soon)

**Reply by DZ8540** [reply]
PR has been sent.

**Reply by dev-passion76** [reply]
Hello
UP this post, I am using Angular with forms for logging in, and AdonisJS V5 with routes for login/logout and users (a route to display a list of users).

The login works fine, and there are no errors with logout. However, the response.clearCookie('adonis-session') doesn't seem to work. After logging out, I can still use the route for users, which is protected by the auth middleware. If I try to manually delete the cookie on the client side, my users' route returns an unauthorized status, so that part is working correctly.

thank for your help

angular code

`import { Component } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CookieService } from 'ngx-cookie-service';




@Component({
  selector: 'lib-form-test-adonis',
  templateUrl: './form-test-adonis.component.html',
  styleUrls: ['./form-test-adonis.component.css'],
})
export class FormTestAdonisComponent {

  httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
  };

  myForm: FormGroup;
  constructor(
    private fb: FormBuilder,
    private http: HttpClient,
    private cookieService: CookieService
  ) {
    this.myForm = this.fb.group({
      email: ['test@gmail.com', [Validators.required, Validators.email]],
      password: [
        'azertyuiopdd',
        [Validators.required, Validators.minLength(8)],
      ],
    });
  }

  login() {
    if (this.myForm.valid) {
      const url = 'http://localhost:3333/login';

      const response: Observable<any> = this.http.post(url, this.myForm.value, {
        withCredentials: true,
      });
      response.subscribe((data) => {
        console.log(data);
      });

    }
  }

  logout() {
    const url = 'http://localhost:3333/logout';

    const response: Observable<any> = this.http.post(url,  {
      withCredentials: false,
    });

    response.subscribe((data) => {
      console.log("je suis logout");
      console.log(data);
    });
  }

  users() {
    const url = 'http://localhost:3333/users';
    const response: Observable<any> = this.http.get(url, { withCredentials: true });

    response.subscribe((data) => {
      console.log(data);
    });
  }`


code Adonis

`Route.get("/users", "AuthController.users").middleware("Auth");

Route.post("/register", "AuthController.register");
Route.post("/login", "AuthController.login");
Route.post("/logout", "AuthController.logout");`

`import type { HttpContextContract } from "@ioc:Adonis/Core/HttpContext";
import User from "App/Models/User";
import LoginValidator from "App/Validators/LoginValidator";
import RegisterValidator from "App/Validators/RegisterValidator";

export default class AuthController {
  public async me({ auth }: HttpContextContract) {
    return auth.user;
  }

  public async register({ request, response }: HttpContextContract) {
    const payload = await request.validate(RegisterValidator);
    await User.create(payload);
    console.log("payload", payload);
    return response.status(200).send({ message: "Register successful" });
  }

  public async login({ auth, request, response }: HttpContextContract) {
    await request.validate(LoginValidator);
    const { email, password } = request.all();
    await auth.use("web").attempt(email, password);

    return response.status(200).send({ message: "Login successful" });
  }

  public async logout({ auth, response }: HttpContextContract) {
    await auth.use("web").logout();

    return response
    .status(202)
    .cookie("adonis-session", "", { httpOnly:true, path: "/", sameSite: false })
    .clearCookie("adonis-session")
    .send({ message: "Logout successful" });
  }

  public async users({ response }: HttpContextContract) {
    const users = await User.all();
    return response.status(200).send(users);
  }
}`

thank for your help

**Reply by RomainLanz** [reply]
@dev-passion76 You must also send the credentials when sending the `logout` request. Otherwise, AdonisJS will not know which session it must revoke.

**Reply by dev-passion76** [reply]
Of course :(

I  think i have already tested but il not sure 
Thanks you 
I 'll tested Tomorrow

**Reply by RomainLanz** [reply]
https://github.com/adonisjs/core/discussions/3532#discussioncomment-7677776

**Comment by dev-passion76** [comment]
hello

with this logout it is same result ??

`  logout() {
    const url = 'http://localhost:3333/logout';

    // Make the POST request
    const response: Observable<any> = this.http.post(url,  {
      withCredentials: true,
    });

    response.subscribe((data) => {
      console.log("je suis logout");
      console.log(data);
    });
  }`

**Reply by RomainLanz** [reply]
Please create another discussion; it will be easier to help you.
Also, if you can provide a repository to reproduce the issue, it would be great. 👍🏻

**Reply by dev-passion76** [reply]
ok, i 'll do it asap
thank

**Reply by dev-passion76** [reply]
> Please create another discussion; it will be easier to help you.
> Also, if you can provide a repository to reproduce the issue, it would be great. 👍🏻 

It is done

https://github.com/adonisjs/core/discussions/4297
